io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplus3em/IoShellyPlus3emImpl.java

```diff
@@ -0,0 +1,249 @@
+	private final CalculateEnergyFromPower calculateConsumptionEnergy = new CalculateEnergyFromPower(this,
+			ElectricityMeter.ChannelId.ACTIVE_CONSUMPTION_ENERGY);
+
+	private final Logger log = LoggerFactory.getLogger(IoShellyPlus3emImpl.class);
+	private final BooleanWriteChannel[] digitalOutputChannels;
+
+	private MeterType meterType = null;
+	private SinglePhase phase = null;
+	private String baseUrl;
+
+	private volatile Timedata timedata;
+
+	@Reference()
+	private BridgeHttpFactory httpBridgeFactory;
+	private BridgeHttp httpBridge;
+
+	public IoShellyPlus3emImpl() {
+		super(OpenemsComponent.ChannelId.values(), ElectricityMeter.ChannelId.values(),
+				DigitalOutput.ChannelId.values(), IoShellyPlus3em.ChannelId.values());
+		this.digitalOutputChannels = new BooleanWriteChannel[] { this.channel(IoShellyPlus3em.ChannelId.RELAY) };
+
+		ElectricityMeter.calculateSumActivePowerFromPhases(this);
+	}
+
+	@Activate
+	protected void activate(ComponentContext context, Config config) {
+		super.activate(context, config.id(), config.alias(), config.enabled());
+		this.meterType = config.type();
+		this.baseUrl = "http://" + config.ip();
+		this.httpBridge = this.httpBridgeFactory.get();
+
+		if (this.isEnabled()) {
+			this.httpBridge.subscribeJsonEveryCycle(this.baseUrl + "/status", this::processHttpResult);
+		}
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		this.httpBridgeFactory.unget(this.httpBridge);
+		this.httpBridge = null;
+		super.deactivate();
+	}
+
+	@Override
+	public BooleanWriteChannel[] digitalOutputChannels() {
+		return this.digitalOutputChannels;
+	}
+
+	@Override
+	public String debugLog() {
+		var b = new StringBuilder();
+		var valueOpt = this.getRelayChannel().value().asOptional();
+		b.append(valueOpt.isPresent() ? (valueOpt.get() ? "ON" : "OFF") : "Unknown");
+		b.append("|");
+		b.append(this.getActivePowerChannel().value().asString());
+
+		return b.toString();
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		if (!this.isEnabled()) {
+			return;
+		}
+
+		switch (event.getTopic()) {
+		case EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE -> this.calculateEnergy();
+		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE -> {
+			this.executeWrite(this.getRelayChannel(), 0);
+		}
+		}
+	}
+
+	private void processHttpResult(JsonElement result, Throwable error) {
+		this._setSlaveCommunicationFailed(result == null);
+		if (error != null) {
+			this._setRelay(null);
+			this._setActivePower(null);
+			this._setActiveProductionEnergy(null);
+			this.logDebug(this.log, error.getMessage());
+			return;
+		}
+
+		try {
+			JsonObject jsonResponse = JsonUtils.getAsJsonObject(result);
+
+			JsonArray relays = JsonUtils.getAsJsonArray(jsonResponse, "relays");
+			if (relays != null && !relays.isEmpty()) {
+				JsonObject relay = JsonUtils.getAsJsonObject(relays.get(0));
+				boolean isOn = JsonUtils.getAsBoolean(relay, "ison");
+				this.getRelayChannel().setNextWriteValue(isOn);
```

Comment:
Why are you setting the write-value here? Sounds very wrong. I'll remove it, but please correct me if I am wrong.